File: Practica1/Backend-python/app/services/notification_service.py
from app.repositories.notification_repository import get_notifications_by_user
from datetime import datetime

def format_time_ago(date):
    diff = (datetime.now() - date).total_seconds()

    if diff < 60:
        return "Hace unos segundos"
    if diff < 3600:
        return f"Hace {int(diff/60)} minutos"
    if diff < 86400:
        return f"Hace {int(diff/3600)} horas"
    return f"Hace {int(diff/86400)} días"


def list_notifications(id_usuario):
    notifications = get_notifications_by_user(id_usuario)

    result = []
    for n in notifications:
        result.append({
            "id_notificacion": n[0],
            "mensaje": n[1],
            "leida": n[2],
            "fecha": n[3],
            "fecha_formateada": format_time_ago(n[3])
        })

    return result

File: Practica1/Backend-Nodejs/src/services/notification.service.js
const notificationRepository = require("../repositories/notification.repository");

function formatTimeAgo(date) {
  const diff = Math.floor((new Date() - new Date(date)) / 1000);

  if (diff < 60) return "Hace unos segundos";
  if (diff < 3600) return `Hace ${Math.floor(diff / 60)} minutos`;
  if (diff < 86400) return `Hace ${Math.floor(diff / 3600)} horas`;
  return `Hace ${Math.floor(diff / 86400)} días`;
}

async function listNotifications(id_usuario) {
  const notifications = await notificationRepository.getNotificationsByUser(id_usuario);

  return notifications.map(n => ({
    id_notificacion: n.id_notificacion,
    mensaje: n.mensaje,
    leida: n.leida,
    fecha: n.fecha,
    fecha_formateada: formatTimeAgo(n.fecha)
  }));
}

module.exports = {
  listNotifications
};